example/functions/create-session.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * This function creates a Stripe Checkout session and returns the session ID
+ * for use with Stripe.js (specifically the redirectToCheckout method).
+ *
+ * @see https://stripe.com/docs/payments/checkout/one-time
+ */
+
+const stripe = require('stripe')(process.env.REACT_APP_STRIPE_API_SECRET);
+
+/*
+ * Product data can be loaded from anywhere. In this case, we’re loading it from
+ * a local JSON file, but this could also come from an async call to your
+ * inventory management service, a database query, or some other API call.
+ *
+ * The important thing is that the product info is loaded from somewhere trusted
+ * so you know the pricing information is accurate.
+ */
+const inventory = require('./data/products.json');
+// const { validateCartItems } = useStripeCart;
+
+const validateCartItems = (inventorySrc, cartItems) => {
+  const validatedItems = Object.keys(cartItems).map((item) => {
+    const product = cartItems[item];
+    const validatedItem = inventorySrc.find(
+      (p) => p.name === cartItems[item].name
+    );
+    return {
+      name: validatedItem.name,
+      amount: validatedItem.price * product.quantity,
+      currency: validatedItem.currency,
+      quantity: product.quantity,
+    };
+  });
```

Comment:
I'm assuming that `cartItems` is an object in this code. This looks like it'd be much more efficient and readable if it was written as a for-loop instead of a `Object.keys().map()` kind of ordeal. For example, you could define the array `validatedItems` above a for-in loop.

```suggestion
  const validatedItems = [];
  for (const item in cartItems) {
    const product = cartItems[item];
    const validatedItem = inventorySrc.find(
      (p) => p.name === cartItems[item].name
    );
    validatedItems.push({
      name: validatedItem.name,
      amount: validatedItem.price * product.quantity,
      currency: validatedItem.currency,
      quantity: product.quantity,
    });
  }
```

Reply:
Also, if `item` is a `sku` it might be more apt to name it `sku`.

Reply:
I agree with the for loop, my only hesitation with naming it `sku` is the fact that if someone wanted to sell subscriptions, it would be a `plan` instead. To be honest, neither sku or plan are required according to the line items spec https://stripe.com/docs/api/checkout/sessions/create#create_checkout_session-line_items

I think because of this ambiguity it would be safer to leave it as "item", although I'm still open to something more descriptive.